contracts/AcrossEventEmitter.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title AcrossEventEmitter
+ * @notice A simple contract that emits events with bytes encoded metadata
+ */
+contract AcrossEventEmitter is ReentrancyGuard {
+    /**
+     * @notice Emitted when metadata is stored
+     * @param data The metadata bytes emitted
+     */
+    event MetadataEmitted(bytes data);
+
+    /**
+     * @notice Prevents native token from being sent to this contract
+     */
+    receive() external payable {
+        revert("Contract does not accept native token");
+    }
+
+    /**
+     * @notice Prevents native token from being sent to this contract via fallback
+     */
+    fallback() external payable {
+        revert("Contract doesn't accept native token");
+    }
+
+    /**
+     * @notice Emits metadata as an event
+     * @param data The bytes data to emit
+     */
+    function emitData(bytes calldata data) external nonReentrant {
+        require(data.length > 0, "Data cannot be empty");
+        require(data.length <= 2048, "Data too large");
```

Comment:
Why limit the size of the data?

Would this protect us from anything?

Reply:
I removed the upper limit [here](https://github.com/across-protocol/contracts/pull/1138/commits/e934edd75d6a7a97584d16d828be9428a7a66d38), but I'd keep the > 0 validation, so that the indexer would not index events with no metadata at all